packages/api-sdk/src/medical/models/common/us-data.ts

```diff
@@ -53,4 +53,15 @@ export enum USState {
   WI = "WI",
   WY = "WY",
 }
-export const usStateSchema = z.nativeEnum(USState);
+
+const USStateMapping = Object.fromEntries(
+  Object.values(USState).map(state => [state, state])
+) as Record<string, USState>;
+
+export const usStateSchema = z
+  .string()
+  .transform(str => str.toUpperCase())
+  .transform(str => USStateMapping[str])
+  .refine(value => value !== undefined, {
+    message: "Invalid US state",
+  });
```

Comment:
Can you try this, please?

```ts
export const usStateSchema = z.preprocess(
  val => (typeof val === "string" ? val.toUpperCase() : val),
  z.nativeEnum(USState)
);
```

Reply:
nice that works too! definetly simpler than my schema!